core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -128,13 +140,16 @@ private[spark] case class ExecutorDeadException(message: String)
 private[spark] class SparkUpgradeException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String],
+    messageParameters: Map[String, String],
     cause: Throwable)
-  extends RuntimeException(SparkThrowableHelper.getMessage(errorClass, errorSubClass.orNull,
-    messageParameters), cause)
-    with SparkThrowable {
+  extends RuntimeException(
+    SparkThrowableHelper.getMessage(errorClass, errorSubClass.orNull, messageParameters), cause)
+  with SparkThrowable {
+
+  override def getMessageParameters: Array[String] = {
+    SparkThrowableHelper.getMessageParameters(errorClass, errorSubClass.orNull, messageParameters)
```

Comment:
@MaxGekk @srielau This line shows up many times in this PR.
Do we consider further refactoring the parameters in `SparkThrowable`? E.g. combine `String[] getMessageParameters` and `String[] getParameterNames` into `Map<String, String> getMessageParameters`

Reply:
+1. These 2 methods were added by https://github.com/apache/spark/pull/36693 and not released yet. We can still change it.

Reply:
I agree. Let's do that in a separate PR.